contracts/babylon/src/contract.rs

```diff
@@ -53,86 +53,82 @@ pub fn instantiate(
 
     // instantiate btc light client contract first
     // It has to be before btc staking and finality contracts which depend on it
-    if let Some(btc_light_client_code_id) = msg.btc_light_client_code_id {
-        let init_btc_lc_msg = if let Some(btc_light_client_msg) = msg.btc_light_client_msg {
-            // If the message is provided, use it
-            btc_light_client_msg
-        } else {
-            // If the message is not provided, use the values in the babylon contract
-            to_json_binary(&btc_light_client::msg::InstantiateMsg {
-                network: msg.network,
-                btc_confirmation_depth: msg.btc_confirmation_depth,
-                checkpoint_finalization_timeout: msg.checkpoint_finalization_timeout,
-                admin: msg.admin.clone(),
-            })?
-        };
-        let init_msg = WasmMsg::Instantiate {
+    let init_btc_lc_msg = if let Some(btc_light_client_msg) = msg.btc_light_client_msg {
+        // If the message is provided, use it
+        btc_light_client_msg
+    } else {
+        // If the message is not provided, use the values in the babylon contract
```

Comment:
if the init msg for each contract can be optional, what's the point of having it?